openvswitch/2.17.1/openvswitch-2.17.1.patch

```diff
@@ -0,0 +1,252 @@
+ OVS_CHECK_USDT
+ OVS_CHECK_NETLINK
++OVS_CHECK_WOLFSSL
+ OVS_CHECK_OPENSSL
+ OVS_CHECK_LIBCAPNG
+ OVS_CHECK_LOGDIR
+diff --git a/lib/dhparams.h b/lib/dhparams.h
+index 0eca99a6f..96b839f12 100644
+--- a/lib/dhparams.h
++++ b/lib/dhparams.h
+@@ -18,6 +18,9 @@
+ #define DHPARAMS_H 1
+ 
+ #include <inttypes.h>
++#ifdef HAVE_WOLFSSL
++#include <wolfssl/options.h>
++#endif
+ #include <openssl/dh.h>
+ 
+ DH *get_dh2048(void);
+diff --git a/lib/stream-ssl.c b/lib/stream-ssl.c
+index f4fe3432e..60159b3fa 100644
+--- a/lib/stream-ssl.c
++++ b/lib/stream-ssl.c
+@@ -24,6 +24,10 @@
+ #include <sys/types.h>
+ #include <sys/socket.h>
+ #include <netinet/tcp.h>
++#ifdef HAVE_WOLFSSL
++#include <wolfssl/options.h>
++#include <openssl/pem.h>
++#endif
+ #include <openssl/err.h>
+ #include <openssl/rand.h>
+ #include <openssl/ssl.h>
+@@ -185,6 +189,10 @@ static unsigned int next_session_nr;
+  * quite a bit. */
+ static struct vlog_rate_limit rl = VLOG_RATE_LIMIT_INIT(10, 25);
+ 
++#ifdef HAVE_WOLFSSL
++static const char *peer_ca_cert_file = NULL;
++#endif
++
+ static int ssl_init(void);
+ static int do_ssl_init(void);
+ static bool ssl_wants_io(int ssl_error);
+@@ -197,6 +205,7 @@ static DH *tmp_dh_callback(SSL *ssl, int is_export OVS_UNUSED, int keylength);
+ static void log_ca_cert(const char *file_name, X509 *cert);
+ static void stream_ssl_set_ca_cert_file__(const char *file_name,
+                                           bool bootstrap, bool force);
++static void stream_ssl_set_peer_ca_cert_file__(const char *file_name);
+ static void ssl_protocol_cb(int write_p, int version, int content_type,
+                             const void *, size_t, SSL *, void *sslv_);
+ static bool update_ssl_config(struct ssl_config_file *, const char *file_name);
+@@ -1004,6 +1013,15 @@ ssl_init(void)
+     return init_status;
+ }
+ 
++#if defined(HAVE_WOLFSSL) && defined(DEBUG_WOLFSSL)
++static void wolfssl_debug_cb(const int log_level, const char *const log_message)
++{
++    (void)log_level;
++
++    VLOG_DBG("wolfSSL debug: %s", log_message);
++}
++#endif
++
+ static int
+ do_ssl_init(void)
+ {
+@@ -1017,6 +1035,10 @@ do_ssl_init(void)
+     SSL_library_init();
+     SSL_load_error_strings();
+ #endif
++#if defined(HAVE_WOLFSSL) && defined(DEBUG_WOLFSSL)
++    wolfSSL_Debugging_ON();
++    wolfSSL_SetLoggingCb(wolfssl_debug_cb);
++#endif
+ 
+     if (!RAND_status()) {
+         /* We occasionally see OpenSSL fail to seed its random number generator
+@@ -1354,13 +1376,8 @@ read_cert_file(const char *file_name, X509 ***certs, size_t *n_certs)
+     return 0;
+ }
+ 
+-
+-/* Sets 'file_name' as the name of a file containing one or more X509
+- * certificates to send to the peer.  Typical use in OpenFlow is to send the CA
+- * certificate to the peer, which enables a switch to pick up the controller's
+- * CA certificate on its first connection. */
+ void
+-stream_ssl_set_peer_ca_cert_file(const char *file_name)
++stream_ssl_set_peer_ca_cert_file__(const char *file_name)
```

Comment:
You can probably get rid of the static declaration above as it isn't called before this implementation and the static keyword needs adding here.